tlslite/tlsrecordlayer.py

```diff
@@ -722,6 +725,31 @@ def _getMsg(self, expectedType, secondaryType=None, constructorType=None):
                                 yield result
                             continue
 
+                    # If we received a heartbeat request ...
+                    if recordHeader.type == ContentType.heartbeat:
+                        # If heartbeat extension was not negotiated, then
+                        # we send unexpected_message fatal alert
+                        if not self.heartbeat:
+                            for result in self._sendError( \
+                                    AlertDescription.unexpected_message,
+                                            "Heartbeat was not negotiated"):
+                                yield result
+
+                        try:
+                            heartbeat_message = Heartbeat().parse(p)
```

Comment:
I found 3 parts talking about when heartbeat messages are to be ignored:
- payload_length of a received HeartbeatMessage is too large
- received HeartbeatResponse message does not contain the expected payload
- it arrives during the handshake
Im not sure, but I think that none of points is this case. I was thinking when parser can fail and he can fail when theres nothing to parse, no payload or different payload and payload length information.